tests/ml_models/test_mcorr.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+"""
+Tests of correctness and compilation for metrics correlation.
+"""
+
+import os
+
+import numpy as np
+import torch
+from sklearn.metrics import precision_score, recall_score
+
+import opensearch_py_ml.ml_models.metrics_correlation.mcorr as mcorr
+
+TESTDATA_FILENAME = os.path.join(
+    os.path.dirname(os.path.abspath("__file__")), "tests", "mcorr_data.pt"
+)
+
+
+def test_level_invariance():
+    """
+    Results should not depend on level of metrics.
+    Test by comparing results on randomly shifted data.
+    """
+    test_data = torch.load(TESTDATA_FILENAME)["mcorr_level"]
+    model = mcorr.MCorr()
+
+    M, _ = test_data.shape
+    events_noshift = model.forward(test_data, max_events=5)
```

Comment:
Should we change `max_events: int = 3` in our function definition then? I remember we had a conversation about checking this default setting?

Reply:
Yes, I think this is a better choice. I can make that change.